lib/AST/ASTContext.cpp

```diff
@@ -563,6 +567,14 @@ void ASTContext::setLazyResolver(LazyResolver *resolver) {
   }
 }
 
+void ASTContext::addLazyParser(LazyMemberParser *lazyParser) {
+  getImpl().lazyParsers.insert(lazyParser);
+}
+
+void ASTContext::removeLazyParser(LazyMemberParser *lazyParser) {
+  getImpl().lazyParsers.erase(lazyParser);
```

Comment:
Maybe `assert` that something was deleted here, so we're sure that all code is following the add/remove discipline?

Reply:
that's a good idea.